docs/rfcs/0007-deployment-chain.md

```diff
@@ -0,0 +1,275 @@
+
+# Motivation
+With the current supporting deployment, our users have to define their application as one of the supporting application's kinds (K8s, Terraform, CloudRun, Lambda, ECS), they can still define multiple PipeCD applications as parts of their application and trigger them one by one, but it's a bit difficult for them to control the deployment chain smoothly.
+
+![image](assets/deployment-chain-kind.png)
+Use-case <1>: chain of deployments contains multiple application cross kinds
+
+Another typical use-case is when users want to build up some kind of chain such as deploy application to the development environment, then if it's done successfully, deploy that application to the staging and then the production environment.
+
+![image](assets/deployment-chain-env.png)
+Use-case <2>: chain of deployments contains application deployed across multiple environments
+
+Or a variant of that requirement, when users want to roll out their applications to their cluster one by one based on its region. As the same below image, the order of the rolling out deployment should be flex, supports both sequential and parallel.
+
+![image](assets/deployment-chain-region.png)
+Use-case <3>: chain of deployments contains application deployed across multiple region
+
+All the above requirements share the same thing in the context, that is: it can be done by users' deployment for their applications as PipeCD applications one by one and make trigger those deployments manually via console or make trigger via pull requests to those application configurations separately, but this all manual stub are tedious and difficult to manage smoothly. With this new __PipeCD deployment chain__ feature, all of those manual steps will be replaced, to keep the good point of using a CD system.
+
+# Detailed design
+The idea is to keep the PipeCD application deployment as a unit of deployment as is, but add __a way to enable users to manipulate a deployment chain based on the state of the last sucessfully run deployment__.
+
+A canonical flow would look as below:
+- Users trigger to run their first application (the first application in their deployment chain) via the web console or pull requests as usual (1)
+- Deploy the first triggered application in the deployment chain (2)
+- After the application is deployed successfully, the piped will send a trigger command to trigger the deployment chain to the control-plane, since all those applications are shared in the same project, this trigger should be valid (3)
+- Control-plane gets deployment events triggered by the first application's piped and makes sync deployment commands for all listed applications (4)
+- Pipeds which handle deployment for all triggered applications fetch sync command and deploy its application once the conditions are satisfied (5)
```

Comment:
```suggestion
1. Users trigger to run their first application (the first application in their deployment chain) via the web console or pull requests as usual
2. Deploy the first triggered application in the deployment chain
3. After the application is deployed successfully, the piped will send a trigger command to trigger the deployment chain to the control-plane, since all those applications are shared in the same project, this trigger should be valid
4. Control-plane gets deployment events triggered by the first application's piped and makes sync deployment commands for all listed applications
5. Pipeds which handle deployment for all triggered applications fetch sync command and deploy its application once the conditions are satisfied
```